### PR TITLE
Differences warning errors

### LINKED ISSUE #595: warnings are counted as errors
Hello,
first of all thanks for this irreplaceable plug-in and for the enhancements given by the last rewriting!

I've a suggestion - please give a look at this screenshot:
schermata 2015-06-24 alle 10 14 39
As you can see in linter warnings are counted as errors (but there are no errors in the file, just warnings). Not a big issue, of course, but it could be an enhancement to distinguish them from real, critical errors.

I can imagine something like "Current File 21/0 | Project 21/0 | X  21 warnings, 0 errors" in the bottom line, or something even more compact by using colors and icons instead of text.

thanks again!
Stefano

installed versions: linter 1.0.2; linter-csslint 0.0.13;

### LINKED ISSUE #597: Keymap is gone
Looks like the very convenient keyboard shortcuts to jump from one "marked as error or warning line" to another are gone, now.

Would it be possible to have them available again, please?

### PR DESCRIPTION
**Pull request by iam4x**
Fixes #595 
Fixes #597

screen shot 2015-06-24 at 13 03 14

### CODE REVIEW

**Comment by iam4x** [comment]
@hd-deman Why in-progress? It should be ready.

**Comment by steelbrain** [comment]
@iam4x Does this show all the different labels as different icons or just Warning and Error and ignores the others?

File: lib/linter-views.coffee [hunk 1]
```diff
@@ -12,19 +12,24 @@ class LinterViews
...
-    @_bottomTabProject.initialize("Project", =>
-      @_changeTab('project')
-    )
-    @_bottomStatus.initialize()
-    @_bottomStatus.addEventListener 'click', ->
+
+    @_bottomErrorStatus = new BottomStatus()
+    @_bottomWarningStatus = new BottomStatus()
+
+    @_bottomTabFile.initialize 'Current File', => @_changeTab 'file'
```
Comment (steelbrain) [inline lib/linter-views.coffee]: the previous code looks cleaner.

Comment (iam4x) [inline lib/linter-views.coffee]:
```
@_bottomTabProject.initialize("Project", =>
  @_changeTab('project')
)
```

Looks cleaner than:

```
@_bottomTabFile.initialize 'Current File', => @_changeTab 'file'
```

I'm not sure, I'm using features from coffeescript.

**Comment by iam4x** [comment]
I don't know all the possible types of messages, I assume there's only `error` and `warning`. I might be wrong.

Any other type will be treated as `error` with current code.

**Comment by steelbrain** [comment]
How about we keep only one indicator and rename it to something other than error?

**Comment by iam4x** [comment]
@steelbrain I'm not sure about that, feels great to have error/warning splitted as @furins asked

**Comment by AsaAyers** [comment]
@iam4x I updated the body of this PR to include `Fixes #595`. That will cause 595 to close as soon as this gets merged

**Comment by iam4x** [comment]
I added documentation about keymaps / shortcuts to Linter commands

**Comment by steelbrain** [comment]
@AsaAyers how do you feel about this? I think we will be occupying a lot of ui if we keep error and warning separate.

**Comment by AsaAyers** [comment]
That's a good point. I'm not sure if we even need the words. My bar is pretty full:

selection_115

Also, I have no idea why I have errors, but they aren't showing up in the count or the labels. I don't know if it's related to this issue or not

**Comment by iam4x** [comment]
@AsaAyers On master you still don't have error count in current file / project?

Use case:

- No errors / warning: show only one label with `√ ` (No text?)
- No errors with warning: Don't show errors but warning
- No warning but errors: Don't show warning but errors

Do we agree?

**Comment by furins** [comment]
@AsaAyers @steelbrain I agree that words may encumber too much the bottom bar. What about using just colors and numbers (I've added also other colors like information and debug info, just to show other labels)? 
example
 
@iam4x what about when you have both warnings and errors?

**Comment by AsaAyers** [comment]
That's a good point. For some reason @steelbrain wants to support an unlimited number of possible types, so this has no bounds on how wide it could be. If we're going to support types everywhere else they should be here too

**Comment by iam4x** [comment]
(@furins Just edited my answer with my missing use case while you were posting)

**Comment by steelbrain** [comment]
We initially had that status indicator to show the user of errors in project, but then we added tabs and it became redundant, right when I was thinking about removing it, someone asked for the feature where they want to be taken to next error or problem if you wanna call it that on click. so I kept it again.

Now adding another one besides it not something I can agree upon. Sorry.

**Comment by steelbrain** [comment]
Just saw that message with colors, I am definately :+1:  on that
but we should drop the commands then

**Comment by furins** [comment]
@iam4x In my (ugly) example: if no warnings/errors/anything the colored areas may collapse. When the code is just "perfect" you may put a green √

**Comment by AsaAyers** [comment]
> but we should drop the commands then

What commands and why?

**Comment by steelbrain** [comment]
@AsaAyers the commands like `linter:next-warning`, `linter:next-info`. We should only have one command to jump to the next option.

**Comment by AsaAyers** [comment]
I agree, maybe `linter:next-message`

**Comment by iam4x** [comment]
I'm okay with it, it removes complexity from commands.

**Comment by steelbrain** [comment]
@iam4x It's gonna be a tiny bit complex to support all the types but I think you should do it, when `LinterViews::_render` is called, it should create a set of errors create counts of all of them, then create html elements for them and replace the old ones with them.

**Comment by steelbrain** [comment]
Also @iam4x do you want to keep working on it or hand it off to someone else?

**Comment by iam4x** [comment]
@steelbrain We can merge this one first, since it works. And make a new one with enhancement, what do you think?

**Comment by k2b6s9j** [comment]
I have a question though, if I use like 'syntax error' as a type will it be marked as an error?

**Comment by AsaAyers** [comment]
What about a compromise? There is currently a bug where it says "Error" but counts everything. Can we simply change the count to be correct, and then add the other message types as an enhancement? Basically for this PR you'd just remove the new `Warning` side of this

**Comment by AsaAyers** [comment]
@k2b6s9j every `type` value on the [API](https://github.com/AtomLinter/Linter/wiki/Linter-API) becomes its own category. If you use `type: 'Syntax Error'` then it becomes a new type unrelated to `Error`. Your package also has to provide the styling for that type, which gets converted to a class named `syntax-error`

**Comment by AsaAyers** [comment]
This came from a tweet I can't find now, but IMO there should be two levels. Warnings are for things that should be fixed, and Errors are for things that never should have been written.

Info is just extra noise

**Comment by iam4x** [comment]
If we stick to warning / issue I can just change a bit the UI to make it smaller and we are good to go :+1:

**Comment by iam4x** [comment]
Made some improvements:

screen shot 2015-06-26 at 14 57 48
screen shot 2015-06-26 at 14 57 59
screen shot 2015-06-26 at 14 58 06

cc @steelbrain @AsaAyers @k2b6s9j

**Comment by steelbrain** [comment]
@iam4x Looks cool. How does it look when it has no errors?

**Comment by iam4x** [comment]
screen shot 2015-06-26 at 16 53 04

Nothing is shown, should we add a green dot?

**Comment by iam4x** [comment]
screen shot 2015-06-26 at 17 20 01

Green dot is live :+1:

**Comment by iam4x** [comment]
TODO:

* Changelog entry
* New gif on `README.md`

**Comment by steelbrain** [comment]
@iam4x how about a tick instead of a green dot, as discussed in slack.

**Comment by iam4x** [comment]
I've tried, it feels weird because others errors type doesn't have any icon. A dot make it smoothier.

**Comment by steelbrain** [comment]
Let me know when it's ready to check out

**Comment by iam4x** [comment]
@steelbrain You can test it.

**Comment by k2b6s9j** [comment]
I like it. Will it show custom types too?

**Comment by steelbrain** [comment]
@k2b6s9j Negative.

**Comment by k2b6s9j** [comment]
Could it? In some of my linters I have a `syntax error` type which is a lighter red. It'd be nice if that were down there too.

**Comment by k2b6s9j** [comment]
Then there is `linter-clang` which has a `fatal error` type which is a darker red.

**Comment by iam4x** [comment]
@k2b6s9j 
https://github.com/AtomLinter/Linter/pull/598#issuecomment-114890793
https://github.com/AtomLinter/Linter/pull/598#issuecomment-114893996

**Comment by iam4x** [comment]
Guys, let's get this merged. I will rebase and then we can close 2 issues.

**Comment by steelbrain** [comment]
/cc @AsaAyers @k2b6s9j @lierdakil
I am neutral on this feature, I mean, It's don't use the the status icon anyway so I'll let those who use this review this PR.

**Comment by lierdakil** [comment]
Screenshots look pretty nice. What message types would be supported in final version?

**Comment by k2b6s9j** [comment]
I think we should move it over to the right of the bar.

**Comment by steelbrain** [comment]
@lierdakil Only `Error` and `Warning`

**Comment by iam4x** [comment]
@lierdakil Like @AsaAyers said we will support split errors / warning (errors = anything else than warnings)

**Comment by lierdakil** [comment]
@k2b6s9j, please don't. It'd be super-awkward on ultrawide monitors.

**Comment by k2b6s9j** [comment]
@devoncarew Thoughts?

**Comment by iam4x** [comment]
Left/Right side isn't in the scope of the PR, it should be in a next issue. This solves the splitting of errors and warning.

**Comment by k2b6s9j** [comment]
I still think that all the different types of issues should be represented, other than that looks good to me.

**Comment by lierdakil** [comment]
I'm not sure about "errors" being "everything else". Personally, I use 'error', 'warning' and 'lint' message levels... lints are not errors. Far from it.

**Comment by iam4x** [comment]
We can add more types support. But the issue was splitting errors / warning :)

**Comment by lierdakil** [comment]
I lean to "add more (all) types", but I guess there's also an option to limit error counter to 'error' type, and 'warning' counter to 'warning' type for now? Maybe?

**Comment by steelbrain** [comment]
@iam4x sorry but I am going to close this one, because the problem was fixed by renaming `Errors` to `Issues` and the other part of the problem will be fixed very soon by having a config to set the scope of this indicator. still :clap: thanks for your efforts

**Comment by iam4x** [comment]
@steelbrain No worries, I had no good purpose to support all types of "issues" can't assign automatically a color to a new type.

**Pull request closed without merging**